File: apps/desktop/src/domain/compatibility.ts
import type { AirtableFieldType } from "./airtable";

export type FieldRestoreSupport =
  | "restorable"
  | "partially_restorable"
  | "metadata_only"
  | "unsupported_for_restore"
  | "manual_action_required";

export interface FieldCompatibilityRule {
  fieldType: AirtableFieldType;
  support: FieldRestoreSupport;
  note: string;
  backupSupport: "full" | "partial" | "metadata_only" | "none";
}

export interface CompatibilitySummary {
  totalFieldTypes: number;
  bySupport: Record<FieldRestoreSupport, number>;
}

File: apps/desktop/src-tauri/src/models/compatibility.rs
use serde::{Deserialize, Serialize};

#[allow(dead_code)]
#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "camelCase")]
pub enum FieldRestoreSupport {
    Restorable,
    PartiallyRestorable,
    MetadataOnly,
    UnsupportedForRestore,
    ManualActionRequired,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct FieldCompatibilityRule {
    pub field_type: String,
    pub support: FieldRestoreSupport,
    pub note: String,
    pub backup_support: String,
}
